compiler/il/OMROpcodes.hpp

```diff
@@ -0,0 +1,13256 @@
+/*******************************************************************************
+ * Copyright (c) 2000, 2020 IBM Corp. and others
+ *
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at http://eclipse.org/legal/epl-2.0
+ * or the Apache License, Version 2.0 which accompanies this distribution
+ * and is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the
+ * Eclipse Public License, v. 2.0 are satisfied: GNU General Public License,
+ * version 2 with the GNU Classpath Exception [1] and GNU General Public
+ * License, version 2 with the OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+
+#ifndef OMR_OPCODEMACROS_INCL
+#define OMR_OPCODEMACROS_INCL
+
+#define FOR_EACH_OPCODE(MACRO) \
+   MACRO(\
+      /* .opcode               = */ TR::BadILOp, \
+      /* .name                 = */ "BadILOp", \
+      /* .properties1          = */ 0, \
+      /* .properties2          = */ 0, \
+      /* .properties3          = */ 0, \
+      /* .properties4          = */ 0, \
+      /* .dataType             = */ TR::NoType, \
+      /* .typeProperties       = */ 0, \
+      /* .childProperties      = */ ILChildProp::NoChildren, \
+      /* .swapChildrenOpCode   = */ TR::BadILOp, \
+      /* .reverseBranchOpCode  = */ TR::BadILOp, \
+      /* .booleanCompareOpCode = */ TR::BadILOp, \
+      /* .ifCompareOpCode      = */ TR::BadILOp, \
+      /* .description          = */ BadILOp = 0, /* illegal op hopefully help with uninitialized nodes */ \
+      /* .simplifierHandler    = */ dftSimplifier, \
+      /* .vpHandler            = */ constrainChildren, \
```

Comment:
Is there a way the comment field could be picked up by doxygen so it could generate the opcode documentation from the opcode definition?

Reply:
@Leonardo2718 both `.opcode`, `.name` and `.description` could be easily substituted by using `.description` (which should rather be named `.enumValue`).

However, there is the first opcode (BadILOp) that I am struggling with because there is a dilemma:

In *OMRILOpCodesEnum.hpp*, we should use `BadILOp = 0`; whereas we need just `BadILOp` for token pasting with simplifier handlers, VP handlers, as well as architecture-specific tree evaluators. If I allow for the former, token pasting results in incorrect macro name, e.g., `BadILOp = 0VPHandler`. If I allow for the latter, a static assert checking `NumILOps` fails.

Reply:
I'd say you leave it as is this for now. I haven't been able to think of an elegant way of doing this yet.